File: app/src/main/java/com/example/forecastapplication/data/db/room/dao/CityDao.java
package com.example.forecastapplication.data.db.room.dao;

import androidx.lifecycle.LiveData;
import androidx.room.Dao;
import androidx.room.Delete;
import androidx.room.Insert;
import androidx.room.OnConflictStrategy;
import androidx.room.Query;
import androidx.room.Update;


import com.example.forecastapplication.data.db.room.entity.City;

import java.util.List;

@Dao
public interface CityDao {

    @Insert(onConflict = OnConflictStrategy.REPLACE)
    void insert(City city);

    @Query("DELETE FROM city_table")
    void deleteAll();

    @Query("SELECT * from city_table ORDER BY id DESC")
    LiveData<List<City>> getAllCity();

    // This method is used to test if table has at least one data.
    @Query("SELECT * from city_table LIMIT 1")
    City[] getAnyCity();

    @Delete
    void deleteCity(City city);

    @Update
    void update(City... city);
}

File: app/src/main/java/com/example/forecastapplication/data/db/room/dao/CityDao.kt
package com.example.forecastapplication.data.db.room.dao

import androidx.lifecycle.LiveData
import androidx.room.*
import com.example.forecastapplication.data.db.room.entity.City

@Dao
interface CityDao {
    @Insert(onConflict = OnConflictStrategy.REPLACE)
    suspend fun insert(city: City?)

    @Query("DELETE FROM city_table")
    fun deleteAll()

    @get:Query("SELECT * from city_table ORDER BY id DESC")
    val allCity: LiveData<List<City?>?>?

    // This method is used to test if table has at least one data.
    @get:Query("SELECT * from city_table LIMIT 1")
    val anyCity: Array<City?>?

    @Delete
    suspend fun deleteCity(city: City?)

    @Update
    suspend fun update(vararg city: City?)
}